src/main/java/com/dropbox/core/http/OkHttp3Requestor.java

```diff
@@ -361,14 +367,14 @@ public void writeTo(BufferedSink sink) throws IOException {
         }
     }
 
-    private static class CountingPipedRequstBody extends PipedRequestBody {
+    private static class ProgressPipedRequstBody extends PipedRequestBody {
```

Comment:
I think you can just wire everything into PipedRequestBody instead of creating a new class. So that we always have same type used as Requestbody.